pkg/controller/storagecluster/storagecluster.go

```diff
@@ -1155,6 +1161,27 @@ func (c *Controller) CreatePodTemplate(
 	return newTemplate, nil
 }
 
+// The aim of this function is to return a good value for MaStorageNodesPerZone with the
+// intention of having at least 3 nodes in the cluster.
+func getDefaultMaxStorageNodesPerZone(zoneMap map[string]uint64) uint32 {
+	numZones := len(zoneMap)
+	switch numZones {
+	case 1:
```

Comment:
help me understand the use case: if there is only one zone with 10 nodes, only 3 of them will be storage nodes? why not let all nodes be storage nodes

Reply:
Choosing the least number was the one with least negative impact. The consensus from the discussions was to start with 3 and then let the customer choose a different number if they choose to do so. The point of this enforcement is to introduce the customer to this option and avoid unnecessary additions of storage node during upgrades.